DeviceSDK/src/main/java/org/astarteplatform/devicesdk/protocol/AstarteInterfaceMapping.java

```diff
@@ -31,7 +33,7 @@ public Type getType() {
   }
 
   protected boolean isTypeCompatible(Type otherType) {
-    return (getType() == otherType);
+    return otherType == getType() || otherType == primitiveArrayType;
```

Comment:
Don't use accessor methods for local fields. 
`return otherType == this.type || otherType == this.primitiveArrayType;`